src/NzbDrone.Core/Indexers/Definitions/MyAnonamouse.cs

```diff
@@ -210,84 +213,96 @@ public class MyAnonamouseRequestGenerator : IIndexerRequestGenerator
         public MyAnonamouseSettings Settings { get; set; }
         public IndexerCapabilities Capabilities { get; set; }
 
-        public MyAnonamouseRequestGenerator()
+        private IEnumerable<IndexerRequest> GetPagedRequests(SearchCriteriaBase searchCriteria)
         {
-        }
+            var term = searchCriteria.SanitizedSearchTerm.Trim();
 
-        private IEnumerable<IndexerRequest> GetPagedRequests(string term, int[] categories)
-        {
-            var qParams = new NameValueCollection
+            var searchType = Settings.SearchType switch
+            {
+                (int)MyAnonamouseSearchType.Active => "active",
+                (int)MyAnonamouseSearchType.Freeleech => "fl",
+                (int)MyAnonamouseSearchType.FreeleechOrVip => "fl-VIP",
+                (int)MyAnonamouseSearchType.Vip => "VIP",
+                (int)MyAnonamouseSearchType.NotVip => "nVIP",
+                _ => "all"
+            };
+
+            var parameters = new NameValueCollection
             {
                 { "tor[text]", term },
+                { "tor[searchType]", searchType },
                 { "tor[srchIn][title]", "true" },
                 { "tor[srchIn][author]", "true" },
-                { "tor[searchType]", Settings.ExcludeVip ? "nVIP" : "all" }, // exclude VIP torrents
                 { "tor[searchIn]", "torrents" },
-                { "tor[hash]", "" },
                 { "tor[sortType]", "default" },
-                { "tor[startNumber]", "0" },
+                { "tor[perpage]", searchCriteria.Limit?.ToString() ?? "100" },
+                { "tor[startNumber]", searchCriteria.Offset?.ToString() ?? "0" },
                 { "thumbnails", "1" }, // gives links for thumbnail sized versions of their posters
-
-                //{ "posterLink", "1"}, // gives links for a full sized poster
-                //{ "dlLink", "1"}, // include the url to download the torrent
                 { "description", "1" } // include the description
-
-                //{"bookmarks", "0"} // include if the item is bookmarked or not
             };
 
-            var catList = Capabilities.Categories.MapTorznabCapsToTrackers(categories);
+            if (Settings.SearchInNarrator)
+            {
+                parameters.Add("tor[srchIn][narrator]", "true");
+            }
+
+            if (Settings.SearchInSeries)
+            {
+                parameters.Add("tor[srchIn][series]", "true");
+            }
+
+            if (Settings.SearchInFilenames)
```

Comment:
Is there any case for always searching in all 3 of these instead of having it user selectable? Just thinking about potential support cases because of varying results. I'm fine leaving it as setting if we think it has value.

Reply:
Nah, we can leave those. Just trying to think about best reliability out of the box. In many cases users won't go look at the settings, or they start toggling them not knowing what they do then come for support. 

Seems like your fixups are best of both worlds here.